src/vehicle_display.cpp

```diff
@@ -120,6 +120,114 @@ nc_color vehicle::part_color( const int p, const bool exact ) const
+
+    for( size_t i = 0; i < parts_id_list.size(); i++ ) {
+
+        const size_t id = static_cast<size_t>( parts_id_list[i] );
+        const vehicle_part &vp = parts[id];
+        //color
+        nc_color sym_clr = c_light_gray;
+        nc_color part_clr = vp.is_broken() ? c_dark_gray : vp.get_base().damage_color();
+        //name
+        std::string partname = vp.name() + " ";
+        //fuel content
+        partname += ( vp.is_fuel_store() && vp.ammo_current() != "null" ) ? fuel_info( vp ) : "";
+        //cargo
+        partname += part_flag( id, "CARGO" ) ? cargo_info( id ) : "";
+        //armor
+        auto sym = armor_info( id );
+        //highlight selected
+        if( static_cast<const int>( i ) == hl ) {
+            sym_clr = hilite( c_light_gray );
+            part_clr = hilite( part_clr );
+        }
+
+        sym.first = colorize( sym.first, sym_clr );
+        sym.second = colorize( sym.second, sym_clr );
+        partname = colorize( partname, part_clr );
+        partname = trim_to_width( ( sym.first + partname + sym.second ), max_width );
+        out_info.push_back( partname );
+    }
+
+    //Covered or not?
+    size_t line_written = ( out_info.size() - old_size );
+    if( line_written > 0 ) {
+
+        const bool is_inside = vpart_position( *this, parts_id_list[0] ).is_inside();
+        std::string desc = is_inside ? _( "Interior" ) : _( "Exterior" );
+
+        const size_t target = out_info.size() - line_written;
+        std::string &line = out_info[target + desc_offset];
+
+        const size_t lenght = static_cast<size_t>( utf8_width( desc ) );
+        const size_t start_pos = lenght > max_width ? max_width : max_width - lenght;
+        const size_t naked_size = remove_color_tags( line ).size();
+
+        if( naked_size > start_pos ) {
+            line = trim_to_width( line, start_pos );
+        } else {
+            std::string padding( ( start_pos - naked_size ), ' ' );
+            line += padding;
+        }
+        line += colorize( desc, c_light_gray );
+    }
+
+    //the label for this location
+    const auto label = vpart_position( *this, veh_part ).get_label();
+    if( label ) {
+        foldstring( out_info, colorize( string_format( "Label: %s", label->c_str() ), c_light_red ),
+                    max_width );
+    }
+
+    return out_info.size() - old_size;
+}
+
+std::string vehicle::fuel_info( const vehicle_part &vp )const
+{
+    return string_format( "(%s)", item::nname( vp.ammo_current() ) );
+}
+
+std::string vehicle::cargo_info( const int id )const
+{
+    //~ used/total volume of a cargo vehicle part
+    return string_format( _( "(vol: %s/%s %s)" ),
+                          format_volume( stored_volume( id ) ),
+                          format_volume( max_volume( id ) ),
+                          volume_units_abbr() );
+}
+
+std::pair<std::string, std::string> vehicle::armor_info( const int id )const
```

Comment:
this should be a vehicle_part function

Reply:
I've moved them, but cargo_info() and armor_info() are currently broken because they rely on functions inside of vehicle and I don't know how to access them.
Anyway to be honest I had moved on to a new project, so I currently don't have the time to put my head into this ^_^'
If anyone suggest the code, I'll accept it, or feel free to take over the branch and do the change.

Reply:
well, probably I can do it myself if I know how to get the vehicle from a vehicle_part

Reply:
Never mind then.  Leave them how you have them and I'll go back and  fix it later.